src/react-vite/12-testing/testing.md

```diff
@@ -38,71 +38,77 @@ React Testing Library’s north star is:
 
 > The more your tests resemble the way your software is used, the more confidence they can give you.
 
-This leads to three [Guiding Principles](https://testing-library.com/docs/guiding-principles/):
-
-> 1. If it relates to rendering components, then it should deal with DOM nodes rather than component instances, and it should not encourage dealing with component instances.
-
 React Testing Library encourages tests that focus on how the user interacts with the application,
-rather than the implementation details of the components. This approach makes the tests more
-resilient to changes in the application’s codebase, as they are less likely to break with
-refactoring if the user interface and behavior remain consistent.
-
-> 2. It should be generally useful for testing the application components in the way the user would use it. We are making some trade-offs here because we’re using a computer and often a simulated browser environment, but in general, utilities should encourage tests that use the components the way they're intended to be used.
-
-The library promotes accessibility in React applications by providing tools and encouraging patterns
-that make it easier to create accessible web interfaces. By using semantic HTML and ARIA roles in
-tests, developers are nudged towards building applications that are more accessible to all users,
-including those using assistive technologies.
-
-> 3. Utility implementations and APIs should be simple and flexible.
-
-React Testing Library is designed to be lightweight and straightforward, avoiding the complexity that
-can sometimes come with extensive testing frameworks. It encourages good testing practices, such as
-avoiding testing the internal state of components and instead focusing on what the user would see
-and do. The library’s API is intentionally limited to encourage tests that align with these principles.
-
-### Rendering and verifying a component in a test
+rather than the implementation details of the components.
 
-Let’s take a look at some code that we added in [Handling User Inputs and
-Forms](./controlled-vs-uncontrolled.html).
+### <span id="rendering-and-verifying">Rendering and verifying a component in a test</span>
```

Comment:
The IDs get created automatically, so is the `span` necessary?

Reply:
My mistake. Fixed.
Need to update the page generation to make one of those little "copy link" widgets on the headers which is what I really want.